packages/core/admin/_internal/node/vite/watch.ts

```diff
@@ -0,0 +1,58 @@
+import { createServer } from 'vite';
+import path from 'node:path';
+import fs from 'node:fs/promises';
+import type { Common } from '@strapi/types';
+import connect from 'koa-connect';
+
+import { mergeConfigWithUserConfig, resolveDevelopmentConfig } from './config';
+
+import type { BuildContext } from '../createBuildContext';
+
+const watch = async (ctx: BuildContext) => {
+  const config = await resolveDevelopmentConfig(ctx);
+  const finalConfig = await mergeConfigWithUserConfig(config, ctx);
+
+  ctx.logger.debug('Vite config', finalConfig);
+
+  const vite = await createServer(config);
+
+  ctx.strapi.server.app.use(connect(vite.middlewares));
+
+  // const serveAdmin: Common.MiddlewareHandler = async (koaCtx, next) => {
+  //   await next();
+
+  //   if (koaCtx.method !== 'HEAD' && koaCtx.method !== 'GET') {
+  //     return;
+  //   }
+
+  //   if (koaCtx.body != null || koaCtx.status !== 404) {
+  //     return;
+  //   }
+
+  //   const url = koaCtx.originalUrl;
+
+  //   // 1. Read index.html
+  //   let template = await fs.readFile(path.resolve(ctx.runtimeDir, 'index.html'), 'utf-8');
+
+  //   // 2. Apply Vite HTML transforms. This injects the Vite HMR client,
+  //   //    and also applies HTML transforms from Vite plugins, e.g. global
+  //   //    preambles from @vitejs/plugin-react
+  //   template = await vite.transformIndexHtml(url, template);
+
+  //   // 5. Inject the app-rendered HTML into the template.
+  //   // const html = template.replace(`<!--ssr-outlet-->`, appHtml);
+  //   koaCtx.type = 'html';
+  //   koaCtx.body = template;
+  // };
+
+  // ctx.strapi.server.routes([
+  //   {
+  //     method: 'GET',
+  //     path: `${ctx.basePath}:path*`,
+  //     handler: serveAdmin,
+  //     config: { auth: false },
+  //   },
+  // ]);
```

Comment:
because we can serve with the middleware, i don't think this is necessary anymore...?